common/models/user.js

```diff
@@ -668,6 +668,45 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    var oldEmail, newEmail, oldId, newId;
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        if (ctx.instance.id !== undefined) {
+          ctx.hookState.id = ctx.instance.id;
+          oldId = ctx.hookState.id;
+        }
+      }
+      if (ctx.data) {
+        ctx.hookState.id = ctx.where.id;
```

Comment:
Also when performing a partial update, the email may not change at all. For example:

```js
user.updateAttributes({ age: user.age + 1 }, cb);
```

You should skip the session-invalidation logic in such case too.